lib/archethic/metrics/metric_client.ex

```diff
@@ -0,0 +1,51 @@
+defmodule ArchEthic.Metrics.MetricClient do
+  @moduledoc """
+  gensever
+  Visit dif
+  """
+  use GenServer
+  @process_name __MODULE__
+  @default_state %{
+    counter: 0
+  }
+  def start_link(_init_arg) do
+    GenServer.start_link(__MODULE__, @default_state, name: @process_name)
+  end
+
+  def init(_state) do
+    {:ok, @default_state}
+  end
+
+  def get_this_node_points() do
+    ArchEthic.Metrics.MetricNodePoller.get_points()
+  end
+
+  def get_network_points() do
+    ArchEthic.Metrics.MetricNetworkPoller.get_points()
+  end
+
+  def monitor() do
+    GenServer.call(@process_name, :monitor)
+  end
+
+  def handle_call(:monitor, {from_pid, _ref}, state) do
+    ArchEthic.Metrics.MetricNodePoller.set_flag()
+    ArchEthic.Metrics.MetricNetworkPoller.set_flag()
+    IO.inspect "metric_client"
+    IO.inspect state
+    _mref = Process.monitor(from_pid)
+    {:reply, :ok, %{state | counter: state.counter + 1}}
```

Comment:
I think it should be better to register the client pid.
Therefore, we will get a list of clients processes.
It's every time good to keep a track of the registered processes to monitor.
Then you want to stop the poller, just check if the processes list is empty.

Reply:
But the data of grow  will grow on large number of user . ??

Reply:
I removed all the the IO.inspect  , this might be an older commit.

Reply:
this is no big brainer, as the clients will be removed once they disconnect